File: cpp/1742. 盒子中小球的最大数量.cpp
class Solution {
public:
    int countBalls(int lowLimit, int highLimit) {
        vector<int> box(60);
        int ans=0;
        for(int i=lowLimit;i<=highLimit;i++){
            int res=0;
            int t=i;
            while(t){
                res+=t%10;
                t/=10;
            }
            box[res]++;
            ans=max(ans,box[res]);
        }
        return ans;
    }
};

File: golang/1742. 盒子中小球的最大数量.go
func countBalls(lowLimit int, highLimit int) (ans int) {
    count:=map[int]int{}
    for i:=lowLimit;i<=highLimit;i++{
        sum:=0
        for x:=i;x>0;x/=10{
            sum+=x%10
        }
        count[sum]++
        ans=max(ans,count[sum])
    }
    return
}
func max(a,b int) int{
    if(a>b) {
        return a
    }
    return b
}